From eadba58294b8d33d3ce18876ace70fe492224811 Mon Sep 17 00:00:00 2001
From: Piotr Szczepanik <piter75@gmail.com>
Date: Tue, 11 Aug 2020 09:33:30 +0200
Subject: [PATCH] Added xt25f32b and xt25f128b SPI NOR flashes from XTX

These flash chips are used by Radxa in ROCK Pi 4.
Based on:
	https://github.com/radxa/kernel/commit/8216f17965de7bc7ced7092aab0e2bfe416838a4
	https://github.com/radxa/kernel/commit/a14f6502e0454a51626e3906f59637ab264bf53e

Signed-off-by: Piotr Szczepanik <piter75@gmail.com>
---
 drivers/mtd/spi-nor/spi-nor.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/mtd/spi-nor/spi-nor.c b/drivers/mtd/spi-nor/spi-nor.c
index 276998ea..cbef5b7f 100644
--- a/drivers/mtd/spi-nor/spi-nor.c
+++ b/drivers/mtd/spi-nor/spi-nor.c
@@ -851,6 +851,10 @@ static const struct flash_info spi_nor_ids[] = {
 	{ "cat25c09", CAT25_INFO( 128, 8, 32, 2, SPI_NOR_NO_ERASE | SPI_NOR_NO_FR) },
 	{ "cat25c17", CAT25_INFO( 256, 8, 32, 2, SPI_NOR_NO_ERASE | SPI_NOR_NO_FR) },
 	{ "cat25128", CAT25_INFO(2048, 8, 64, 2, SPI_NOR_NO_ERASE | SPI_NOR_NO_FR) },
+
+	/* XTX SPI Nor Flash */
+	{ "xt25f32b", INFO(0x0b4016, 0, 64 * 1024, 64, SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
+	{ "xt25f128b", INFO(0x0b4018, 0, 64 * 1024, 256, SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
 	{ },
 };
 
-- 
2.17.1

